python/pyspark/sql/streaming/stateful_processor.py

```diff
@@ -77,6 +78,58 @@ def clear(self) -> None:
         self._value_state_client.clear(self._state_name)
 
 
+class ListState:
+    """
+    Class used for arbitrary stateful operations with transformWithState to capture single value
+    state.
+
+    .. versionadded:: 4.0.0
+    """
+
+    def __init__(
+        self, list_state_client: ListStateClient, state_name: str, schema: Union[StructType, str]
+    ) -> None:
+        self._list_state_client = list_state_client
+        self._state_name = state_name
+        self.schema = schema
+
+    def exists(self) -> bool:
+        """
+        Whether list state exists or not.
+        """
+        return self._list_state_client.exists(self._state_name)
+
+    def get(self) -> Iterator["PandasDataFrameLike"]:
```

Comment:
@HeartSaVioR I'm keeping the `Iterator["PandasDataFrameLike"]` for `get()` for now since I feel it's easier for users to use `PandasDataFrameLike` as a input parameter for `put()` and `appendList()`. It's better to keep it consistent for `get()` as well. But let me know if you have different ideas, thanks!

Reply:
cc. @anishshri-db

Reply:
We decided to go with Row for get() in ValueState, right? I'm not sure we want to give different UX for ValueState vs ListState. It's conceptually just `List[ValueState]`, not a totally different once.

Reply:
I circled back with `Iterator["PandasDataFrameLike"]` mainly because we provide the same parameter in `handleInputRows` where users can directly use `rows: Iterator["PandasDataFrameLike"]` to construct list state for `put()` and `appendList()`. 
```
def handleInputRows(
        self, key: Any, rows: Iterator["PandasDataFrameLike"]
    ) -> Iterator["PandasDataFrameLike"]:
```
If we only support `Iterator[Row]` for `put()` and `appendList()`, users might need extra transformation to achieve the same. @anishshri-db do you know if that's a common use case?

Reply:
We consider Spark SQL's Row as the base type of the state. ValueState followed the underlying implementation, and I expect all types to follow the same.

I'm OK to accept PandasDataFrameLike, but it should be secondary type of support (should be `Union[Row, PandasDataFrameLike]`) and we need to take care of conversion to finally store them to "Row". (We need to error out when given pandas DataFrame does not fit with the state schema.)

Reply:
@HeartSaVioR OK, if we'd like to use `Iterator[Row]` in the interfaces, I think it's better to use the same for both `put()` and `get()` instead of `get()` only. Let me see how I could adjust that.

Reply:
Updated to Iterator[Row]